draft-ietf-httpbis-header-structure.md

```diff
@@ -853,17 +845,17 @@ NOTE: This algorithm parses both Integers ({{integer}}) and Floats ({{float}}),
 7. While input_string is not empty:
    1. Let char be the result of consuming the first character of input_string.
    2. If char is a DIGIT, append it to input_number.
-   3. Else, if type is "integer" and char is ".", append char to input_number and set type to "float".
+   3. Else, if type is "integer" and char is ".", append char to input_number and set type to "decimal".
    4. Otherwise, prepend char to input_string, and exit the loop.
    5. If type is "integer" and input_number contains more than 15 characters, fail parsing.
-   6. If type is "float" and input_number contains more than 16 characters, fail parsing.
+   6. If type is "decimal" and input_number contains more than 16 characters, fail parsing.
 8. If type is "integer":
    1. Parse input_number as an integer and let output_number be the product of the result and sign.
    2. If output_number is outside the range −999,999,999,999,999 to 999,999,999,999,999 inclusive, fail parsing.
 9. Otherwise:
    1. If the final character of input_number is ".", fail parsing.
-   2. If the number of characters after "." in input_number is greater than six, fail parsing.
-   2. Parse input_number as a float and let output_number be the product of the result and sign.
+   2. If the number of characters after "." in input_number is greater than three, fail parsing.
```

Comment:
Your algorithm doesn't explicitly mark the placement of the "." or count characters.  I think that's OK, it's just imprecise.

Reply:
It's not clear whether `1234567890123.0` or `0034567890123.0` should fail.

Reply:
Right now they don't. Separate issue?

Reply:
Yeah, they both fail now with 087d6b1, and the second one is a bit silly anyway.